client/lib/query-args/index.ts

```diff
@@ -0,0 +1,59 @@
+import { getQueryArgs } from '@wordpress/url';
+import page from 'page';
+
+function getRelativeUrlWithParameters( queryArgs: object ): string {
+	const urlParams = new URLSearchParams( window.location.search );
+	let returnPath = window.location.pathname;
+
+	if ( queryArgs ) {
+		for ( const [ key, value ] of Object.entries( queryArgs ) ) {
+			if ( '' !== value ) {
+				urlParams.set( key, value );
+			} else {
+				urlParams.delete( key );
+			}
+		}
+
+		returnPath += '?' + urlParams.toString();
+	}
+
+	return returnPath;
+}
+
+function useQueryArgs() {
+	function setQueryArgs( queryArgs: object ) {
+		const searchWithoutBaseURL = getRelativeUrlWithParameters( queryArgs );
+
+		page.replace( searchWithoutBaseURL );
```

Comment:
For clarity I don't know if it is or not, just noticed the difference.

Reply:
On my tests, both of them generated the same result from the user's perspective. Do you have any preference here?

Reply:
I'm not familiar with [page.js](https://github.com/visionmedia/page.js/blob/master/index.js) enough to know, is `page.replace()` replacing current search in history with the latest one? Does `page()` push history on top or is it different some other way?

In the case of executing a search on every key stroke it might make sense to replace as you go but for search on enter I think it makes sense to preserve the full browser forward/back history. (i.e. use `page()` instead of `page.replace()`)

Reply:
>  page.replace() replacing current search in history with the latest one? 
Seems to be actually doing that per [this](https://github.com/visionmedia/page.js/blob/4f9991658f9b9e3de9b6059bade93693af24d6bd/index.js#L278-L287) and [this](https://github.com/visionmedia/page.js/blob/4f9991658f9b9e3de9b6059bade93693af24d6bd/index.js#L278-L287) which calls [history.replaceState](https://developer.mozilla.org/en-US/docs/Web/API/History/replaceState).

> Does page() push history on top or is it different some other way?
It seems to be the case.

> In the case of executing a search on every key stroke it might make sense to replace as you go but for search on enter I think it makes sense to preserve the full browser forward/back history. (i.e. use page() instead of page.replace())

Totally agree on this one I think we should always use page for this one.